src/kayenta/report/detail/graph/chartjs/graph.tsx

```diff
@@ -1,17 +1,34 @@
 import * as React from 'react';
-import { Chart } from 'chart.js';
+import { Chart, ChartLegendItem } from 'chart.js';
 
 import { IMetricSetPairGraphProps } from '../metricSetPairGraph.service';
 import { buildChartConfig } from './chartConfigFactory';
 
+export interface ChartLegendItemCallback {
+    (e: MouseEvent, legendItem: ChartLegendItem): void;
+};
+
 export default class ChartJSGraph extends React.Component<IMetricSetPairGraphProps> {
   private canvas: HTMLCanvasElement;
   public chart: Chart;
 
+  public legendOnClickHideAxis : ChartLegendItemCallback = (e: MouseEvent, legendItem: ChartLegendItem): void => {
+    var xAxes = this.chart.config.options.scales.xAxes;
```

Comment:
```suggestion
    const xAxes = this.chart.config.options.scales.xAxes;
```

Reply:
just to double check as I do not know fully how typescript works. i am modifying the underlying object in the map closure. is this acceptable in tsx?

Reply:
yep! all good. The [`const`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const) assignment is only referring to the binding of the `xAxes` scalar to this particular object. Mutating references inside the data structure bound to `xAxes` is still legal in the world of `const`. An example of something that **would** be an exception:

```ts
const xAxes = this.chart.config.options.scales.xAxes;

// xAxes is now getting bound to a different reference, not allowed with const 
xAxes = { another: 'object' };
```

In the cases where you **do** need to reassign a local binding to a different reference, you can use [`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let). [`var`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var) is the original local binding operator in JS and uses function scoping — not block scoping like you'd find in most C-family languages — so it can have some behaviors that are unpredictable to users familiar with block scoping.